### PR TITLE
18769: Configure CI (with push) for Boilerplate repository

### PR DESCRIPTION
**Pull request by kamilwysocki**
Added CircleCI config.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+version: 2
+jobs:
+  build-and-test:
+    docker:
+      - image: uxpin2/node8:1.0
```
Comment (Eyjafjallajokull) [inline .circleci/config.yml:4]:
```suggestion
      - image: uxpin2/node8:8.1.2-1
```

**Review by Eyjafjallajokull — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "@babel/core": "^7.1.2",
     "@babel/preset-env": "^7.1.0",
     "@babel/preset-react": "^7.0.0",
-    "@uxpin/merge-cli": "^1.1.0",
+    "@uxpin/merge-cli": "1.1.0-dev.2",
```
Comment (kamilwysocki) [inline package.json]: ⏰

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          command: npm install
+      - run:
+          name: "Run tests"
+          command: npm run test
+      - run:
+          name: "Check ESLint"
+          command: ./node_modules/eslint/bin/eslint.js --ext js,jsx src --ignore-pattern *.tests.js
+      - run:
+          name: 'Push to UXPin'
+          command: ./node_modules/@uxpin/merge-cli/bin/uxpin-merge push --webpack-config ./webpack.config.js --wrapper ./src/UXPinWrapper/UXPinWrapper.js --uxpin-domain uxpin.com --token $TOKEN
```
Comment (michalstocki) [inline .circleci/config.yml]:
```suggestion
          command: ./node_modules/@uxpin/merge-cli/bin/uxpin-merge push --webpack-config ./webpack.config.js --wrapper ./src/UXPinWrapper/UXPinWrapper.js --uxpin-domain uxpin.com
```

`--token` is not needed as we read from `UXPIN_AUTH_TOKEN` Env variable by default

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  build-and-test:
+    docker:
+      - image: uxpin2/node8:1.0
+        auth:
+          username: $DOCKER_USER
+          password: $DOCKER_PASS
+    working_directory: ~/project/
+    steps:
+      - checkout
+      - run:
```
Comment (michalstocki) [inline .circleci/config.yml:11]: Let's remove "Login to NPM". Boilerplate is a **public** repository. We shouldn't install anything that is private.

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+version: 2
+jobs:
+  build-and-test:
```
Comment (michalstocki) [inline .circleci/config.yml]:
```suggestion
  build:
```
Workflow is not needed here.

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+          name: "Run tests"
+          command: npm run test
+      - run:
+          name: "Check ESLint"
+          command: ./node_modules/eslint/bin/eslint.js --ext js,jsx src --ignore-pattern *.tests.js
+      - run:
+          name: 'Push to UXPin'
+          command: ./node_modules/@uxpin/merge-cli/bin/uxpin-merge push --webpack-config ./webpack.config.js --wrapper ./src/UXPinWrapper/UXPinWrapper.js --uxpin-domain uxpin.com --token $TOKEN
+
+workflows:
```
Comment (michalstocki) [inline .circleci/config.yml]: We can remove workflow configuration as we have only one job.

File: .circleci/config.yml (hunk 3 above)
Comment (michalstocki) [inline .circleci/config.yml]:
```suggestion
          command: ./node_modules/.bin/uxpin-merge push --webpack-config ./webpack.config.js --wrapper ./src/UXPinWrapper/UXPinWrapper.js
```

- `--token` is not needed as we read from `UXPIN_AUTH_TOKEN` Env variable by default
- `--uxpin-domain` is not needed as `uxpin.com` is a default value

**Review by michalstocki — CHANGES_REQUESTED** [review]

**Review by michalstocki — APPROVED** [review]

File: .circleci/config.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      - checkout
+      - run:
+          name: "Install dependencies"
+          command: npm install
+      - run:
+          name: "Run tests"
+          command: npm run test
+      - run:
+          name: "Check ESLint"
+          command: ./node_modules/eslint/bin/eslint.js --ext js,jsx src --ignore-pattern *.tests.js
```
Comment (jaroslawsawicki) [inline .circleci/config.yml]: it could be run as `npm run check`

Comment (kamilwysocki) [reply .circleci/config.yml]: You mean - moving this whole command to package.json and firing it here? Cause there's no `npm run check` task yet

**Review by jaroslawsawicki — APPROVED** [review]

**Pull request merged**